ui/src/components/navbar/navbar.tsx

```diff
@@ -1,23 +1,59 @@
 'use client';
-import { Group, Button, Box } from '@mantine/core';
+import { Group, Button, Box, Avatar, Anchor, Input } from '@mantine/core';
 import { IconLovelace } from '@/components/misc/icon-lovelace';
+import { IconSearch } from '@tabler/icons-react';
 import classes from './navbar.module.css';
 
-export default function Navbar() {
+export enum NavbarStatus {
+  HOME = 'HOME',
+  LOGGED = 'LOGGED',
+  AUTH = 'AUTH',
+}
+
+function NavbarAuthOptions() {
+  return null;
+}
+
+function NavbarLoggedOptions() {
+  return (
+    <Group visibleFrom="sm" className={classes.menu}>
+      <Group>
+        <Anchor href="/problemset" c={'gray.0'} fw={700}>
+          Problemas
+        </Anchor>
+        <Anchor href="/groupset" c={'dark.1'} fw={600}>
+          Turmas
+        </Anchor>
+      </Group>
+      <Input placeholder="Buscar" leftSection={<IconSearch size={16} />} />
+      <Avatar></Avatar>
+    </Group>
+  );
+}
+
+function NavbarUnloggedOptions() {
+  return (
+    <Group visibleFrom="sm">
+      <Button component="a" href="/register" variant="default">
+        Cadastrar
+      </Button>
+      <Button component="a" href="/login" variant="gradient" autoContrast>
+        Entrar
+      </Button>
+    </Group>
+  );
+}
+
+export default function Navbar({ status }: { status?: NavbarStatus }) {
   return (
     <Box>
       <header className={classes.header}>
```

Comment:
Usar nav na navbar ao invés de header, apenas sugestão abaixo, alterar para que faça sentido
```suggestion
      <nav className={classes.header}>
```